packages/date/src/TimezonePickerInput.tsx

```diff
@@ -1,32 +1,77 @@
-import React, { ChangeEvent } from 'react';
-import { zoneOffsets, defaultZoneOffset } from './utils/zoneOffsets';
+import React, { useCallback, useEffect, useState } from 'react';
 
-import { Select } from '@contentful/f36-components';
+import { Autocomplete } from '@contentful/f36-components';
+
+import allTimezones, { Timezone } from './utils/timezones';
+import { defaultZoneOffset } from './utils/zoneOffsets';
+
+import dayjs from 'dayjs';
+
+import utc from 'dayjs/plugin/utc';
+import timezone from 'dayjs/plugin/timezone';
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
 
 export type TimezonepickerProps = {
   disabled: boolean;
   onChange: (value: string) => void;
   value?: string;
 };
+
 export const TimezonepickerInput = ({
   disabled,
   onChange,
   value = defaultZoneOffset,
 }: TimezonepickerProps) => {
+  const localTimezone = dayjs.tz.guess();
+  const defaultTimezone =
+    value && value !== dayjs.tz(undefined, localTimezone).format('Z')
+      ? allTimezones.find(
+          (timezone: Timezone) => dayjs.tz(undefined, timezone.ianaName).format('Z') === value
+        )
+      : allTimezones.find((timezone: Timezone) => timezone.ianaName === localTimezone);
+
+  const [filteredTimezones, setFilteredTimezones] = useState(allTimezones);
+  const [userInput, setUserInput] = useState(defaultTimezone as Timezone);
```

Comment:
```ts
const defaultTimezone =
```

will run on each re-render, but we only need it for the state so I would recommend doing this ( I added some [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to make the code shorter)

```suggestion
    const [filteredTimezones, setFilteredTimezones] = useState(allTimezones);
  const [userInput, setUserInput] = useState<Timezone>(() => {
    const localTimezone = dayjs.tz.guess();

    const timezone =
      value === dayjs.tz(undefined, localTimezone).format('Z')
        ? allTimezones.find(({ ianaName }: Timezone) => ianaName === localTimezone)
        : allTimezones.find(
            ({ ianaName }: Timezone) => dayjs.tz(undefined, ianaName).format('Z') === value
          );

    return timezone as Timezone;
  });
```
----